core/src/test/java/io/fabric8/maven/core/handler/ContainerHandlerTest.java

```diff
@@ -0,0 +1,324 @@
+package io.fabric8.maven.core.handler;
+
+import io.fabric8.kubernetes.api.model.ContainerPort;
+import io.fabric8.maven.core.config.ResourceConfig;
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.maven.core.config.VolumeConfig;
+import io.fabric8.maven.docker.config.BuildImageConfiguration;
+import io.fabric8.maven.docker.config.ImageConfiguration;
+import io.fabric8.utils.Strings;
+import mockit.Mocked;
+import org.apache.maven.project.MavenProject;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class ContainerHandlerTest {
+
+    @Mocked
+    EnvVarHandler envVarHandler;
+
+    @Mocked
+    ProbeHandler probeHandler;
+
+    private List<Container> containers;
+
+    @Test
+    public void getContainersTest(){
+        MavenProject project = new MavenProject();
+
+        ContainerHandler handler = new ContainerHandler(project, envVarHandler, probeHandler);
+        ResourceConfig config = new ResourceConfig.Builder()
+                .imagePullPolicy("IfNotPresent")
+                .controllerName("testing")
+                .withReplicas(5)
+                .build();
+
+        List<String> ports = new ArrayList<>();
+        ports.add("8080");
+        ports.add("9090");
+
+        List<String> tags = new ArrayList<>();
+        tags.add("latest");
+        tags.add("test");
+
+        BuildImageConfiguration buildImageConfiguration1 = new BuildImageConfiguration.Builder().
+                ports(ports).from("fabric8/maven:latest").cleanup("try").tags(tags).compression("gzip").build();
+
+        ImageConfiguration imageConfiguration1 = new ImageConfiguration.Builder().
+                name("test").alias("test-app").buildConfig(buildImageConfiguration1).registry("docker.io").build();
+
+        BuildImageConfiguration buildImageConfiguration2 = new BuildImageConfiguration.Builder().
+                ports(ports).from("fabric8/").cleanup("try").tags(tags)
+                .compression("gzip").dockerFile("testFile").dockerFileDir("/demo").build();
+
+        ImageConfiguration imageConfiguration2 = new ImageConfiguration.Builder().
+                name("test").alias("test-app").buildConfig(buildImageConfiguration2).registry("docker.io").build();
+
+        List<ImageConfiguration> images = new ArrayList<>();
+        images.add(imageConfiguration1);
+        images.add(imageConfiguration2);
+
+        containers = handler.getContainers(config, images);
+
+        assertEquals(2,containers.size());
```

Comment:
Can we assert other important container attributes as well? e.g. container name

Reply:
@hrishin getImageName function is using ImageConfiguration as input to derive the image name, in ImageConfiguration class, there are only two relevant fields `name` and `registry`. ImageConfiguration class is from docker-maven-plugin. You can check the getImageName() code [here](https://github.com/fabric8io/fabric8-maven-plugin/blob/master/core/src/main/java/io/fabric8/maven/core/handler/ContainerHandler.java#L96)

Reply:
@hrishin This has been resolved in the new commit. Thanks